db/migrate/20210820135643_create_category_usages.rb

```diff
@@ -0,0 +1,9 @@
+class CreateCategoryUsages < ActiveRecord::Migration[5.2]
```

Comment:
What are your thoughts an implementing an entirely new model for this feature when we have the ahoy tracking in place? I'm not entirely convinced yet so am open to understanding the thought process or pros for doing it this way :) @bradjohnson92008 @JoshingYou1

Reply:
Yeah.. I didn't think of that when i heard the requirements for this.  If there is an advantage to using ahoy here  I'm pretty sure I can make those changes pretty quickly.  It might be more work to extract the "top 6" in ahoy...  I just created the category_usage table.... which has the category_id, key (for preventing multiple updates on same request due to turbolinks) and timestamps ... so grabbing the top 6 is just an easy group by statement..  anyway - majority rules - if you and Josh want it in the ahoy tables - I understand.

Reply:
I'm leaning more toward using the ahoy events, and I'll explain why. We should be able to get the top 6 through querying the Ahoy::Event records, as well as the 90 day interval. To prevent multiple updates on the same request, we can check the ahoy table for two events that are very close in time (probably `ms`) and make sure they have the same category name (or query string). I think the only real advantage to using ahoy over a new table is the fact that we have to create a new table, as opposed to utilizing one we already have in the DB. Just my 2 🪙

Reply:
refactoring for ahoy.tracking.